core/src/main/java/org/infinispan/interceptors/impl/TxInterceptor.java

```diff
@@ -858,4 +717,168 @@ protected E getNextFromIterator() {
          return returnedValue;
       }
    }
+
+   private class TxKeyCacheSet extends AbstractDelegatingKeyCacheSet<K, V> {
+      private final boolean isRemoteIteration;
+      private final InvocationContext rCtx;
+
+      TxKeyCacheSet(VisitableCommand rCommand, CacheSet<K> set, boolean isRemoteIteration,
```

Comment:
I'd like ```isRemoteIteration``` to be the last arg so it looks identical to ```TxEntryCacheSet``` constructor. Let me do this refactoring myself and merge.